internal/controller/ctrlpkg/util.go

```diff
@@ -1,7 +1,29 @@
 package ctrlpkg
 
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+)
+
 func IsSameResource(a, b Package) bool {
 	return a.GetName() == b.GetName() &&
 		a.GroupVersionKind() == b.GroupVersionKind() &&
 		a.GetNamespace() == b.GetNamespace()
 }
+
+func SpecHash(pkg Package) (string, error) {
```

Comment:
I think it would be fair to move this function to the api package, so it can be called as `pkg.GetSpec().Hashed()` or something like that